content/movie-review-pt2.md

```diff
@@ -0,0 +1,520 @@
+# Movie Review Program Pt2
+
+# Lesson Objectives
+
+- Describe the process of creating a new account using a Program Derived Address (PDA)
+- Use seeds to derive a PDA
+- Use the space required by an account to calculate the amount of rent (in lamports) a user must allocate
+- Use a Cross Program Invocation (CPI) to initialize an account with a PDA as the address of the new account
+- Explain how to update the data stored on a new account
+
+# TL;DR
+
+- Creating an account requires that we calculate the space and rent to allocate for a new account
+- A Program Derived Address (PDA) is derived from a program ID and an optional list of seeds
+- Creating a new account requires that perform a Cross Program Invocation (CPI) to the `create_account` instruction on the System Program
+- Updating the data field on an account requires that we serialize (convert to byte array) the data into the account
+
+# Overview
+
+In the previous lesson we went over how to, deserialize instruction data, deserialize account data, and print messages to the program log.
+
+In this lesson we will build upon what we have already done. The next step is to create a new account to store the movie review data from the user.
+
+## Account State
+
+In order to store data on a new account, we must first define a struct that serves as the format for the data field of the account. These structs are stored in a `state.rs` file and are used to serialize data into an account and deserialize data from an existing account.
```

Comment:
I want to tweak the intro to this section just a bit. This is good context but I think we can do more.

> Under the hood, account data is represented as a byte array, just as it is with instruction data. Since we'll be using account data to manage our program's state, one of the first things we can do to make managing state easier is to represent it as a Rust data type. 

> While this type should reflect the structure of your data, for most use cases a simple struct is sufficient. For example, a note-taking program that stores notes in their own accounts would likely have data for a title, body, and maybe an ID of some kind. We could create a struct to represent that as follows:

```rust
struct NoteState {
    title: String,
    body: String,
    id: u64
}
```